mobile/adaptation.html

```diff
@@ -36,6 +36,7 @@
 
   <div data-feature="CSS-based adaptation">
     <p><a data-featureid="css-size-adjust">CSS Mobile Text Size Adjustment</a> lets text adapt to zoomed parts of a page.</p>
+    <p><a data-featureid="viewportapi">Visual Viewport API</a> provides developers with a way to interact with the properties of the visual viewport.</p>
```

Comment:
Two things:

1. That proposal does not fit in the "CSS-based adaptation" bucket because, well, it's a JavaScript API ;)
2. I suggest to be clearer on the distinction between visual and layout viewports.

Something like:

```html
<p data-feature="JS-based adaptation">Most mobile devices feature on-screen keyboards for user input. They also commonly let the user zoom in on a page. Both of these features preserve the <em>layout viewport</em> (what the web page uses when laying out its user interface), so as not to clutter the screen with <code>position: fixed</code> elements. The <a data-featureid="viewportapi">Visual Viewport API</a> provides developers with a way to query and interact with the properties of the <em>visual viewport</a> (the box on the page that the user can currently see), and to be notified when these properties change.</p>
```

Reply:
But it's in the CSSOM...

Reply:
Events are tied to CSSOM, but my point here is that, from a developer perspective, this is not CSS-based adaptation: the developer is not going to write CSS to use the Visual Viewport API, she's going to write JS.

CSS-based adaptation is something like CSS Media Queries: the developer will write CSS directives such as `@media all and (min-width: 320px)`. No JS involved.